Environment/Environment.cpp

```diff
@@ -502,32 +502,31 @@ int setclipboard(lua_State *L) {
     if (Environment::get_singleton()->get_instrumentation_status()) {
         printf("--- setclipboard invoked ---\r\n");
     }
-    const char *newClipboard = lua_tostring(L, 1);
+    luaL_checkany(L, 1);
+    const char* data = luaL_tolstring(L, 1, NULL);
+    lua_pop(L, 1);
 
-    if (!OpenClipboard(nullptr)) {
-        lua_pushnil(L);
-        return 0;
-    }
+    HWND hwnd = GetDesktopWindow();
 
+    OpenClipboard(hwnd);
     EmptyClipboard();
-    const auto allocSize = strlen(newClipboard) + 1;
-    const HGLOBAL newMemory = GlobalAlloc(GMEM_MOVEABLE, allocSize);
-    if (newMemory == nullptr) { // Rip windows, malloc
-        CloseClipboard();
-        return 0;
-    }
 
-    if (!GlobalLock(newMemory)) {
+    HGLOBAL hg = GlobalAlloc(GMEM_MOVEABLE, strlen(data) + 1);
+
+    if (!hg)
+    {
         CloseClipboard();
-        GlobalFree(newMemory);
         return 0;
     }
 
-    std::memcpy(newMemory, newClipboard, allocSize);
-    GlobalUnlock(newMemory);
-    SetClipboardData(CF_TEXT, newMemory);
+    memcpy(GlobalLock(hg), data, strlen(data) + 1);
```

Comment:
https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-globallock

Global lock may return a nullptr, this will cause UB, fix.